### PR TITLE
8300693: Lower the compile threshold and reduce the iterations of warmup loop in VarHandles tests

### PR DESCRIPTION
**Pull request by mlchung**
`java/lang/invoke/VarHandles` tests run with C1, C2 and tiered compilations and the test cases are executed in the warm up loop with 20000 iterations to verify C1, C2 intrinsics.  Default Tier4CompileThreshold is 15000.

This PR proposes to scale the compile threshold to 0.1 such that the warm up loop can be reduced to 2000 iterations.  This will speed up the test execution time.

```
Before:
make test-only TEST=open/test/jdk/java/lang/invoke/VarHandles  341.06s user 14.65s system 563% cpu 1:03.14 total

After:
make test-only TEST=open/test/jdk/java/lang/invoke/VarHandles  234.38s user 13.08s system 535% cpu 46.218 total
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8300693](https://bugs.openjdk.org/browse/JDK-8300693): Lower the compile threshold and reduce the iterations of warmup loop in VarHandles tests

### Reviewers
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - **Reviewer**) ⚠️ Review applies to [0e003531](https://git.openjdk.org/jdk/pull/12104/files/0e003531940f657ffeba714572f055d0d33e8468)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Paul Sandoz](https://openjdk.org/census#psandoz) (@PaulSandoz - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12104/head:pull/12104` \
`$ git checkout pull/12104`

Update a local copy of the PR: \
`$ git checkout pull/12104` \
`$ git pull https://git.openjdk.org/jdk pull/12104/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12104`

View PR using the GUI difftool: \
`$ git pr show -t 12104`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12104.diff">https://git.openjdk.org/jdk/pull/12104.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/invoke/VarHandles/VarHandleTestAccessBoolean.java [hunk 1]
```diff
@@ -280,6 +280,8 @@ public Object[][] accessTestCaseProvider() throws Exception {
         return cases.stream().map(tc -> new Object[]{tc.toString(), tc}).toArray(Object[][]::new);
     }
 
+    // CompileThresholdScaling is set to 0.1 and ITERS to 2000 to reduce
+    // the test execution
```
Comment (dholmes-ora) [inline test/jdk/java/lang/invoke/VarHandles/VarHandleTestAccessBoolean.java:284]: Wouldn't this be clearer as an `@comment` in the test header prior to the `@run` statements that use the variables?

**Review by JornVernee — APPROVED** [review]

File: test/jdk/java/lang/invoke/VarHandles/VarHandleTestAccessBoolean.java (hunk 1 above)
Comment (mlchung) [reply test/jdk/java/lang/invoke/VarHandles/VarHandleTestAccessBoolean.java]: Good suggestion.  Updated.

**Review by dholmes-ora — APPROVED** [review]
Looks god. Thanks.

**Review by PaulSandoz — APPROVED** [review]

**Pull request closed without merging**